web/src/common/Loading.svelte

```diff
@@ -0,0 +1,50 @@
+<script lang="ts">
+  export let loading;
+  export let progress = null;
+
+  let startTime: number | undefined = undefined;
+  let taskName: string | undefined = undefined;
+  $: {
+    if (loading && !startTime) {
+      // Start timing when loading begins
+      startTime = Date.now();
+      taskName = loading;
+    } else if (!loading && startTime) {
+      // Calculate and log duration when loading ends
```

Comment:
I copied this component from utils. The main change I made is that we now log the loading time whenever the loading banner is shown. It's not super precise, but since we're only ever showing the loading banner for relatively slow things, it's accurate enough to be helpful.

Reply:
Nice!

Eventually, I want to figure out how to plumb progress bar messages back from the WASM backend and really beef up this component. That's a long ways off though. This is a great step forward in the meantime.